SmartDeviceLink/SDLFuelRange.h

```diff
@@ -1,35 +1,96 @@
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the SmartDeviceLink Consortium Inc. nor the names of
+ * its contributors may be used to endorse or promote products derived
+ * from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
 #import "SDLFuelType.h"
+#import "SDLRPCStruct.h"
+#import "SDLComponentVolumeStatus.h"
+#import "SDLCapacityUnit.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
 /**
- *  Describes the distance a vehicle can travel with the current level of fuel.
+ * @since SDL 5.0.0
  */
 @interface SDLFuelRange : SDLRPCStruct
 
 /**
- *  The vehicle's fuel type
- *
- *  Optional
+ * @param type - type
+ * @param range - @(range)
+ * @param level - @(level)
+ * @param levelState - levelState
+ * @param capacity - @(capacity)
+ * @param capacityUnit - capacityUnit
+ * @return A SDLFuelRange object
  */
+- (instancetype)initWithType:(nullable SDLFuelType)type range:(float)range level:(float)level levelState:(nullable SDLComponentVolumeStatus)levelState capacity:(float)capacity capacityUnit:(nullable SDLCapacityUnit)capacityUnit;
+
 @property (nullable, strong, nonatomic) SDLFuelType type;
 
 /**
- *  The estimate range in KM the vehicle can travel based on fuel level and consumption.
- *
- *  Optional, Float, 0 - 10,000
+ * The estimate range in KM the vehicle can travel based on fuel level and consumption.
+ * {"default_value": null, "max_value": 10000.0, "min_value": 0.0}
  */
 @property (nullable, strong, nonatomic) NSNumber<SDLFloat> *range;
 
+/**
+ * The relative remaining capacity of this fuel type (percentage).
+ * {"default_value": null, "max_value": 1000000.0, "min_value": -6.0}
+ *
+ * @since SDL 6.2.0
+ */
+@property (nullable, strong, nonatomic) NSNumber<SDLFloat> *level;
+
+/**
+ * The fuel level state
+ *
+ * @since SDL 6.2.0
+ */
```

Comment:
If the property is `nullable`, you should specific `Optional` in the comment scheme